File: app/src/main/java/com/group15/fitnesstracker/db/Nutrition_Recording.java
package com.group15.fitnesstracker.db;

import java.util.Date;

import androidx.room.Entity;
import androidx.room.PrimaryKey;

@Entity
public class Nutrition_Recording {
    @PrimaryKey
    public int recording_id;

    public Date date;

    public double calories;

    public macronutrient macronutrient;
}

class macronutrient {
    public double protein;
    public double carbohydrate;
    public double fat;
}

File: app/src/main/java/com/group15/fitnesstracker/db/NutritionRecording.kt
package com.group15.fitnesstracker.db

import java.util.Date

import androidx.room.Entity
import androidx.room.ForeignKey
import androidx.room.PrimaryKey

@Entity(foreignKeys = [ForeignKey(entity = User::class, parentColumns = ["id"], childColumns = ["userId"])])
data class NutritionRecording (
        @PrimaryKey var recordingId: Int = 0,
        var userId: Int,
        var date: Date = Date(),
        var calories: Double?,
        var protein: Double?,
        var carbohydrate: Double?,
        var fat: Double?
)